keyboards/th96/keymaps/default/keymap.c

```diff
@@ -0,0 +1,87 @@
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_P7, KC_P8, KC_P9, KC_PMNS, 
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT, KC_P4, KC_P5, KC_P6, KC_PPLS, 
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_P1, KC_P2, KC_P3, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_RALT, MO(2), KC_LEFT, KC_DOWN, KC_RGHT, KC_P0, KC_PDOT, KC_ENT
+    ),
+
+    [2] = LAYOUT_all(
+		RGB_TOG,KC_BRID, KC_BRIU, LGUI(KC_TAB), LGUI(KC_Z), KC_MAIL, LGUI(KC_D), KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, KC_NO, KC_NO, KC_NO, KC_MSTP, KC_BRIU, 
+		KC_NO,  KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, RGB_TOG, KC_NO, KC_NO, KC_BRID, 
+		KC_NO,  KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, RGB_MOD, KC_NO, KC_NO, KC_NO,   KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO,  KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO,   KC_NO, KC_NO, KC_NO, 
+		KC_NO,  KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, RGB_VAI, KC_NO,   KC_NO, KC_NO, 
+		KC_NO,  0x5d61, KC_NO, KC_NO, KC_NO, KC_NO, RGB_SPI, RGB_VAD, RGB_SPD, KC_NO, KC_NO, KC_NO
+    ),
+
+    [3] = LAYOUT_all(
+		KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO
+    )
+
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+	palSetPadMode(GPIOC,  0, PAL_MODE_OUTPUT_PUSHPULL);
+    switch (get_highest_layer(state)) {
+        case 0:
+            palClearPad(GPIOC, 0);
+            break;
+        case 1:
+            palSetPad(GPIOC, 0);
+            break;
+		case 2:
+            palClearPad(GPIOC, 0);
+            break;
+        case 3:
+			palClearPad(GPIOC, 0);//亮
+            break;
+        default:
+            break;
+    }
+  return state;
+}
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if(clockwise){
+    uint16_t keycode = dynamic_keymap_get_keycode(biton32(layer_state),0,17);
+    tap_code(keycode);
+  }
+  else{
+    uint16_t keycode = dynamic_keymap_get_keycode(biton32(layer_state),1,17);
+    tap_code(keycode);
+  }
+  return true;
+}
```

Comment:
This hack isnt accepted any more, <https://docs.qmk.fm/#/feature_encoders?id=encoder-map>

Reply:
removed default keymap